File: backends/ts-hono/src/domain/client/valueObjects.ts
/**
 * クライアントドメイン バリューオブジェクトモジュール。
 *
 * @author Satoshi Nagashiba <satoshi.nagashiba@gmail.com>
 */

/** クライアント一覧レスポンス用 VO。 */
export interface ClientListItem {
  id: number;
  name: string;
  identifier: string;
  status: number;
  startedAt: Date | null;
  stoppedAt: Date | null;
  createdAt: Date | null;
  updatedAt: Date | null;
}

/** クライアント詳細レスポンス用 VO。 */
export interface ClientDetailVo {
  id: number;
  name: string;
  identifier: string;
  postCode: string;
  pref: string;
  city: string;
  address: string;
  building: string;
  tel: string;
  email: string;
  status: number;
  fingerprint: string | null;
  startedAt: Date | null;
  stoppedAt: Date | null;
  createdAt: Date | null;
  updatedAt: Date | null;
}

/** クライアント登録結果 VO。メール送信・通知配信に必要なフィールドを含む。 */
export interface ClientStoreResultVo {
  id: number;
  name: string;
  identifier: string;
  email: string;
  token: string;
}

/** クライアント登録入力 VO（リポジトリ INSERT 用）。 */
export interface ClientStoreVo {
  name: string;
  identifier: string;
  postCode?: string;
  pref?: string;
  city?: string;
  address?: string;
  building?: string;
  tel?: string;
  email?: string;
}

/** クライアント更新入力 VO（リポジトリ UPDATE 用）。 */
export interface ClientUpdateVo {
  name?: string;
  postCode?: string;
  pref?: string;
  city?: string;
  address?: string;
  building?: string;
  tel?: string;
  email?: string;
  status?: number;
}

/** QRコード返却用 VO。 */
export interface ClientQrVo {
  identifier: string;
  deeplinkUrl: string;
}

/** スマホアプリ向けクライアント情報 VO。 */
export interface ClientInfoVo {
  identifier: string;
  name: string;
  status: number;
}

/** 利用開始結果 VO。 */
export interface ClientStartVo {
  accessToken: string;
}

File: backends/python-fastapi/app/domain/client/value_objects.py
"""
クライアントドメイン バリューオブジェクトモジュール。

Author: Satoshi Nagashiba <satoshi.nagashiba@gmail.com>
"""
from dataclasses import dataclass
from datetime import datetime
from typing import Optional


@dataclass(frozen=True)
class ClientListItem:
    """クライアント一覧レスポンス用バリューオブジェクト。"""

    id: int
    name: str
    identifier: str
    status: int
    started_at: Optional[datetime]
    stopped_at: Optional[datetime]
    created_at: Optional[datetime]
    updated_at: Optional[datetime]


@dataclass(frozen=True)
class ClientDetailVo:
    """クライアント詳細レスポンス用バリューオブジェクト。"""

    id: int
    name: str
    identifier: str
    post_code: str
    pref: str
    city: str
    address: str
    building: str
    tel: str
    email: str
    status: int
    fingerprint: Optional[str]
    started_at: Optional[datetime]
    stopped_at: Optional[datetime]
    created_at: Optional[datetime]
    updated_at: Optional[datetime]


@dataclass(frozen=True)
class ClientStoreResultVo:
    """クライアント登録結果バリューオブジェクト。メール送信・通知配信に必要なフィールドを含む。"""

    id: int
    name: str
    identifier: str
    email: str
    token: str


@dataclass(frozen=True)
class ClientQrVo:
    """QRコードデータバリューオブジェクト。"""

    identifier: str
    deeplink_url: str


@dataclass(frozen=True)
class ClientInfoVo:
    """スマホアプリ向けクライアント情報バリューオブジェクト。"""

    identifier: str
    name: str
    status: int


@dataclass(frozen=True)
class ClientStartVo:
    """利用開始結果バリューオブジェクト。"""

    access_token: str
